### ISSUE TITLE
ClientCredsSpotify allows requests before authenticated and panics with 'Rspotify not authenticated'

### ISSUE THREAD

**Issue by kaleidawave** [ramsayleung/rspotify#333]
**Describe the bug**

The following example internally panics with `'Rspotify not authenticated' rspotify-0.11.5\src\clients\base.rs:100:14`

```rust
use rspotify::{clients::BaseClient, ClientCredsSpotify, Credentials};

#[tokio::main]
async fn main() {
    const CLIENT_ID: &str = "...";
    const CLIENT_SECRET: &str = "...";
    let mut client = ClientCredsSpotify::new(Credentials::new(CLIENT_ID, CLIENT_SECRET));

    let results = client
        .playlist_items_manual(
            &"37i9dQZF1DXcBWIGoYBM5M".parse().unwrap(),
            None,
            None,
            Some(5),
            None,
        )
        .await;

    eprintln!("{:#?}", results);
}
```

Turns out there first needs to be a `client.request_token().await.unwrap();` call after creating the client. I feel there should be a intermediate struct so only authenticated clients can call `.playlist_items_manual`. Then the type system mistake could have caught the mistake I made.

Thanks 🙌

**Comment by Spanfile** [comment]
Library code should never panic when the user makes a mistake (in this case, not filling in the authentication information), it should return an error and let the user deal with it however they want. In a similar vein, when the access token eventually expires after an hour and a call returns 401, ideally the library should refresh the token itself and retry the call.

**Comment by marioortizmanero** [comment]
Yeah, you're right. I'll work on this after we merge some of the PRs we have pending. About the token refreshing, we already implement that, but it's opt-in, so it can still happen because not everyone wants to refresh tokens automatically.
